homeassistant/components/media_player/firetv.py

```diff
@@ -195,60 +178,132 @@ def source_list(self):
         """Return a list of running apps."""
         return self._running_apps
 
+    @adb_decorator(override_available=True)
     def update(self):
         """Get the latest date and update device state."""
-        self._state = {
-            'idle': STATE_IDLE,
-            'off': STATE_OFF,
-            'play': STATE_PLAYING,
-            'pause': STATE_PAUSED,
-            'standby': STATE_STANDBY,
-            'disconnected': STATE_UNKNOWN,
-        }.get(self._firetv.state, STATE_UNKNOWN)
-
-        if self._state not in [STATE_OFF, STATE_UNKNOWN]:
-            self._running_apps = self._firetv.running_apps
-            self._current_app = self._firetv.current_app
-        else:
+        # Check if device is disconnected.
+        if not self._available:
             self._running_apps = None
             self._current_app = None
 
+            # Try to connect
+            self.firetv.connect()
+            self._available = True
```

Comment:
How do we know that connection succeeded?

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/17767/commits/d1e57425b6025981e7206d1f494b588bcf4b097c